### PR TITLE
docs: Add page about pandas booleans

### PR DESCRIPTION
**Pull request by MarcoGorelli**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues

- Related issue #\<issue number\>
- Closes #\<issue number\>

## Checklist

- [ ] Code follows style guide (ruff)
- [ ] Tests added
- [ ] Documented the changes

### CODE REVIEW

File: docs/pandas_like_concepts/boolean.md [hunk 1]
```diff
@@ -6,6 +6,22 @@ For example, if you do `nw.col('a')*2`, then:
...
+import pandas as pd
+import polars as pl
+import pyarrow as pa
+
+data = {"a": [1.4, None, 4.2]}
+print("pandas output")
+print(nw.from_native(pd.DataFrame(data)).with_columns(b=nw.col("a") * 2).to_native())
+print("\nPolars output")
+print(nw.from_native(pl.DataFrame(data)).with_columns(b=nw.col("a") * 2).to_native())
+print("\nPolars output")
```
Comment (FBruzzesi) [inline docs/pandas_like_concepts/boolean.md:21]:
```suggestion
print("\nPyArrow output")
```

File: docs/pandas_like_concepts/boolean.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+- for all backends except pandas, null values are preserved
+- for pandas, this depends on the dtype backend:
+    - for PyArrow dtypes and pandas nullable dtypes, null
+      values are preserved
+    - for the classic NumPy dtypes, null values are typically
+      filled in with `False`.
+
+pandas is generally moving towards nullable dtypes, and they
+may become the default in the future, so we hope that the
```
Comment (FBruzzesi) [inline docs/pandas_like_concepts/boolean.md:37]: I tried checking the pandas roadmap, but nothing explicitly states this. Is there anything to support the statement?

**Review by FBruzzesi — COMMENTED** [review]
Definitly nice to have this page, so that we can link it in docstrings as well if needed!
I left two comments :)

File: docs/pandas_like_concepts/boolean.md (hunk 2 above)
Comment (MarcoGorelli) [reply docs/pandas_like_concepts/boolean.md:37]: there's https://github.com/pandas-dev/pandas/pull/58988/files

**Comment by MarcoGorelli** [comment]
thanks for your reviews 🙏

**Pull request merged**